src/core/environment.go

```diff
@@ -194,24 +213,32 @@ func (env *EnvironmentType) GetHostname() string {
 }
 
 func (env *EnvironmentType) GetSystemUUID() string {
-	ctx := context.Background()
-	defer ctx.Done()
+	res, err, _ := singleflightGroup.Do(GetSystemUUIDKey, func() (interface{}, error) {
+		var err error
+		ctx := context.Background()
+		defer ctx.Done()
 
-	if env.IsContainer() {
-		containerID, err := env.GetContainerID()
-		if err != nil {
-			log.Errorf("Unable to read docker container ID: %v", err)
-			return ""
+		if env.IsContainer() {
+			containerID, err := env.GetContainerID()
+			if err != nil {
+				return "", fmt.Errorf("unable to read docker container ID: %v", err)
+			}
+			return uuid.NewMD5(uuid.NameSpaceDNS, []byte(containerID)).String(), err
 		}
-		return uuid.NewMD5(uuid.NameSpaceDNS, []byte(containerID)).String()
-	}
 
-	hostInfo, err := host.InfoWithContext(ctx)
+		hostID, err := host.HostIDWithContext(ctx)
+		if err != nil {
+			log.Infof("Unable to read host id from dataplane, defaulting value. Error: %v", err)
+			return "", err
+		}
+		return uuid.NewMD5(uuid.Nil, []byte(hostID)).String(), err
+	})
 	if err != nil {
-		log.Infof("Unable to read host id from dataplane, defaulting value. Error: %v", err)
+		log.Warnf("unable to set hostname due to %v", err)
```

Comment:
(Optionally follow logging convention)
```suggestion
		log.Warnf("Unable to set hostname due to: %v", err)
```